elasticdl/python/tests/pserver_servicer_test.py

```diff
@@ -0,0 +1,100 @@
+import unittest
+from concurrent import futures
+
+import grpc
+import tensorflow as tf
+from google.protobuf import empty_pb2
+
+from elasticdl.proto import elasticdl_pb2_grpc
+from elasticdl.python.common.constants import GRPC
+from elasticdl.python.ps.parameters import Parameters
+from elasticdl.python.ps.servicer import PserverServicer
+
+
+class PserverServicerTest(unittest.TestCase):
+    @staticmethod
+    def create_server(
+        parameters,
+        grads_to_wait,
+        optimizer,
+        lr_staleness_modulation,
+        use_async,
+        port,
+    ):
+        server = grpc.server(
+            futures.ThreadPoolExecutor(max_workers=64),
+            options=[
+                ("grpc.max_send_message_length", GRPC.MAX_SEND_MESSAGE_LENGTH),
+                (
+                    "grpc.max_receive_message_length",
+                    GRPC.MAX_RECEIVE_MESSAGE_LENGTH,
+                ),
+            ],
+        )
+
+        pserver_servicer = PserverServicer(
+            parameters,
+            grads_to_wait,
+            optimizer,
+            lr_staleness_modulation=lr_staleness_modulation,
+            use_async=use_async,
+        )
+
+        elasticdl_pb2_grpc.add_PserverServicer_to_server(
+            pserver_servicer, server
+        )
+        server.add_insecure_port("[::]:{}".format(port))
+        return server
+
+    @staticmethod
+    def create_stub(port):
+        addr = "localhost:%d" % port
+        channel = grpc.insecure_channel(
+            addr,
+            options=[
```

Comment:
Maybe move this to a constant in this file so it's reused

Reply:
Moved to class attributes for reuse.